data/tutorials/guides/0tt_05_s_expressions.md

```diff
@@ -0,0 +1,268 @@
+# Std.sexp_of_int 0;;
+- : Type.t = Sexp.Atom "0"
+# Std.sexp_of_float 1.2;;
+- : Type.t = Sexp.Atom "1.2"
+# Std.sexp_of_string "hello world";;
+- : Type.t = Sexp.Atom "hello world"
+```
+
+These functions can also be combined with the polymorphic ones available for lists and arrays:
+```ocaml
+# Std.(sexp_of_list sexp_of_int [1; 2; 3]);;
+- : Type.t =
+Sexp.List
+ [Sexp.Atom "1"; Sexp.Atom "2"; Sexp.Atom "3"]
+
+# Std.(sexp_of_array sexp_of_int [|3; 2; 1|]);;
+- : Type.t =
+Sexp.List
+ [Sexp.Atom "3"; Sexp.Atom "2"; Sexp.Atom "1"]
+```
+
+By combining these functions, you can write your own `sexp_of_t` functions for your own types (we will see later how to generate them automatically).
+
+### Converting S-expressions Into OCaml Types
+
+Similarly, it is possible to convert S-expressions into the corresponding OCaml types by using the predefined `t_of_sexp` functions:
+```ocaml
+# Std.int_of_sexp (Sexp.Atom "0");;
+- : int = 0
+# Std.float_of_sexp (Sexp.Atom "1.2");;
+- : float = 1.2
+# Std.string_of_sexp (Sexp.Atom "hello world");;
+- : string = "hello world"
+```
+
+Here again, we can combine these functions with the ones defined for polymorphic data types such as:
+```ocaml
+# Std.list_of_sexp Std.int_of_sexp (Sexp.List [Sexp.Atom "1"; Sexp.Atom "2"; Sexp.Atom "3"]);;
+- : int list = [1; 2; 3]
+# Std.array_of_sexp Std.int_of_sexp (Sexp.List [Sexp.Atom "3"; Sexp.Atom "2"; Sexp.Atom "1"]);;
+- : int array = [|3; 2; 1|]
+```
+
+### Setting Up `ppx_sexp_conv`
+
+[`ppx_sexp_conv`](https://github.com/janestreet/ppx_sexp_conv) is a library and a [PPX rewriter](https://ocaml.org/docs/metaprogramming#ppx-rewriters) that allows conversions between S-expressions and OCaml types.
+
+First, `ppx_sexp_conv` needs to be installed:
+```shell
+$ opam install ppx_sexp_conv
+```
+
+Then, to use `ppx_sexp_conv` in your project, you must list it as a dependency and as a preprocessor in your `dune` file:
+```lisp
+(library
+  (name <your_lib>)
+ (preprocess
+  (pps ppx_sexp_conv))
+ (libraries ppx_sexp_conv <your_dependencies>))
+```
+
+Finally, to add the predefined converters to the scope, you must open the `Sexplib.Std` module.
+
+### Automatically Generate Conversion Functions
+
+`ppx_sexp_conv` will automatically generate conversion functions for the selected OCaml types.
+
+These functions are of the form:
+- `sexp_of_*` to convert from an OCaml type to a S-expression
+- `*_of_sexp` to convert from a S-expression to an OCaml type
+
+For example, `sexp_of_int` and `int_of_sexp` are generated for basic the `int` type. These functions will be specifically named after the OCaml type they allow users to convert to and from.
+
+To automatically generate such functions for your own type, you must annotate the type definition with an attribute: `[@@deriving sexp]`, such as:
+
+```ocaml
+# type color = Red | Blue | Green [@@deriving sexp];;
+type color = Red | Blue | Green
+val color_of_sexp : Sexp.t -> color = <fun>
+val sexp_of_color : color -> Sexp.t = <fun>
```

Comment:
I only get 

`type color = Red | Blue | Green` 

The other two lines aren't there (but I didn't edit the `dune` file. Is that why?

Reply:
I edited the `dune` file in the ocaml.org directory, but it still didn't work. (Not sure I did it right...just added this to the bottom of the file:

```
(library
  (name <your_lib>)
 (preprocess
  (pps ppx_sexp_conv))
 (libraries ppx_sexp_conv <your_dependencies>))
```

Unless I'm supposed to fill in `<your_lib>` with a specific library and `<your_dependencies>` with specific dependencies, but I have no idea which ones. Will your readers know?

Reply:
@gpetiot, Cuit & I looked at this together, and he's getting the same result I am. He determined there's something missing / not working in the code. The heart of the problem is in `[@@deriving sexp];;` in li, 231 as well as lines 250 & 256.